src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js

```diff
@@ -187,6 +187,11 @@
         },
 
         setDate: function(date) {
+            if(date == null || date == '') {
+                this.updateModel(null, null);
```

Comment:
I think you can just do `if (!date) {` because both null and '' will equate to false with truthiness.  I just tested in Chrome Console and both evaluate to false as does undefined.

Reply:
Done